packages/python/plotly/plotly/express/_doc.py

```diff
@@ -180,8 +180,12 @@
         "Values from this column or array_like appear in bold in the hover tooltip.",
     ],
     hover_data=[
-        colref_list_type,
-        colref_list_desc,
+        "list of str or int, or Series or array-like, or dict",
+        "Either a list of names of columns in `data_frame`, or pandas Series,",
+        "or array_like objects",
+        "or a dict with column names as keys, with values True (for default formatting)",
```

Comment:
A few questions:

1. The intent here is that for a scatter, `hover_data=dict(y=False)` would still leave the x in the hovertemplate, right? The implied default is `True`?
2. I imagine if that I wanted a non-data-frame column with custom formatting, I could do something like `custom_data=[ [1,2,3] ], hover_data=dict(custom_data_0 = ".2f"), labels=dict(custom_data_0="my thing")`?

I wonder if we shouldn't allow a tuple instead of just a string value, so people could provide named data in hover like `hover_data={"my thing": (".2f", [1,2,3])}`...

Here are the cases I can imagine:

1. I want to hide a column: `False`
2. I want to format an existing, named column: `".2f"`
3. I want to provide a new named column: not possible now with just `hover_data`
3. I want to provide a new named column and formatting data: not possible now with just `hover_data`

Reply:
interesting, I'll explore the tuple idea.

Reply:
Does it seem like a too-complex signature to accept:

1. Booleans for on/off
2. strings for formatting
3. list-like for data
4. tuples for formatting and data

?

Reply:
awesome!